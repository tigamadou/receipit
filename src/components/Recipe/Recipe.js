import React from 'react';
import PropTypes from 'prop-types';
import {
  Link,
} from 'react-router-dom';
import Fade from 'react-reveal/Fade';
import Color from '../../styles/modules/Colors.module.scss';
import Font from '../../styles/modules/Font.module.scss';
import Card from '../../styles/modules/Card.module.scss';

const Recipe = ({ data }) => (
  <Fade>
    <div className={`card ${Card.withOverlay}`} style={{ backgroundImage: `url(${data.strMealThumb})` }}>
      <Link to={`/recipe/${data.idMeal}`}>
        <div className="content">
          <div className="overlay">
            <div>
              <h2 className={`title ${Color.white} ${Font.is_xl}`}>{data.strMeal}</h2>
            </div>
          </div>
        </div>
      </Link>
    </div>
  </Fade>
);

Recipe.defaultProps = {
  data: null,
};

Recipe.propTypes = {
  data: PropTypes.instanceOf(Object),
};
export default Recipe;
