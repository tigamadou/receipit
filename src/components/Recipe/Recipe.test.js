import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import Recipe from './Recipe';

const data = {
  strMealThumb: 'thumb',
  idMeal: '1',
  strMeal: 'Meal name',

};
describe('Recipe component tests', () => {
  test('renders Recipe component ', () => {
    render(<Router><Recipe data={data} /></Router>);
    expect(screen.getByText(data.strMeal)).toBeInTheDocument();
  });

  test('should match with snapshot', () => {
    const tree = renderer
      .create(<Router><Recipe data={data} /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
