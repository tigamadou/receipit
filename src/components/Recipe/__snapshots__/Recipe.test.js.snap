// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component tests should match with snapshot 1`] = `
<div
  className="card cover withOverlay"
  style={
    Object {
      "backgroundImage": "url(thumb)",
    }
  }
>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="overlay"
      >
        <div>
          <h2
            className="title white is_xl"
          >
            Meal name
          </h2>
        </div>
      </div>
    </div>
  </a>
</div>
`;
